whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/MessageService.java

```diff
@@ -37,46 +38,68 @@ public MessageService(
         this.emailStatusDao = emailStatusDao;
     }
 
-    public MessageInfo getBouncedMessageInfo(final MimeMessage message) throws MessagingException, IOException {
+    public EmailMessageInfo getBouncedMessageInfo(final MimeMessage message) throws MessagingException, IOException {
         return bouncedMessageParser.parse(message);
     }
-    public MessageInfo getUnsubscribedMessageInfo(final MimeMessage message) throws MessagingException, IOException {
+    public EmailMessageInfo getUnsubscribedMessageInfo(final MimeMessage message) throws MessagingException {
         return unsubscribeMessageParser.parse(message);
     }
 
-    public void verifyAndSetAsUndeliverable(final MessageInfo message){
-        final String email = outgoingMessageDao.getEmail(message.messageId());
+    public void verifyAndSetAsUndeliverable(final EmailMessageInfo message){
+        final List<String> outgoingEmail = outgoingMessageDao.getEmails(message.messageId());
 
-        if (isIncorrectMessage(message, email)){
+        if (isNotValidMessage(message, outgoingEmail)){
             return;
         }
 
-        LOGGER.debug("Undeliverable message-id {} email {}", message.messageId(), message.emailAddress());
-        emailStatusDao.createEmailStatus(email, EmailStatus.UNDELIVERABLE);
+        LOGGER.debug("Undeliverable message-id {} email {}", message.messageId(), StringUtils.join(message.emailAddresses(), ", "));
+        message.emailAddresses().forEach(email -> emailStatusDao.createEmailStatus(email, EmailStatus.UNDELIVERABLE));
     }
 
-    public void verifyAndSetAsUnsubscribed(final MessageInfo message){
-        final String email = outgoingMessageDao.getEmail(message.messageId());
+    public void verifyAndSetAsUnsubscribed(final EmailMessageInfo message){
+        if (message.emailAddresses() != null && message.emailAddresses().size() > 1){
+            LOGGER.error("This can not happen, unsubscribe with multiple recipients");
```

Comment:
Warning would be better to avoid so many sentry alerts

Reply:
Warnings are also reported in sentry. Error has higher priority so better keep at warning.

Reply:
Done